inst/examples/joint_rpca_example.qmd

```diff
@@ -0,0 +1,122 @@
+---
+title: "Joint RPCA Reproducible Example"
+format: html
+editor: visual
+---
+
+```{r}
+options(warn = -1)
+# Load user-defined functions
+source("../R/joint_rpca.R")
+source("../R/joint_optspace_helper.R")
+source("../R/joint_optspace_solve.R")
+source("../R/optspace_helper.R")
+source("../R/transform_helper.R")
+source("../R/transform.R")
+source("../R/rclr_transform.R")
+source("../R/mask_value_only.R")
+source("../R/rpca_table_processing.R")
+source("../R/joint_rpca_utils.R")
+
+# TESTING THE JOINT RPCA FUNCTION
+
+#set up synthetic data
+set.seed(123)
+
+#create shared samples
+samples <- paste0("Sample", 1:10)
+features1 <- paste0("F1_", 1:8)
+features2 <- paste0("F2_", 1:10)
+features3 <- paste0("F3_", 1:6)
+
+#generate three synthetic tables
+table1 <- matrix(rpois(80, lambda = 25), nrow = length(features1),
+                 dimnames = list(features1, samples))
+table2 <- matrix(rpois(100, lambda = 30), nrow = length(features2),
+                 dimnames = list(features2, samples))
+table3 <- matrix(rpois(60, lambda = 35), nrow = length(features3),
+                 dimnames = list(features3, samples))
+
+#introduce a few NAs
+table1[sample(length(table1), 5)] <- NA
+table2[sample(length(table2), 6)] <- NA
+table3[sample(length(table3), 4)] <- NA
+
+#sample metadata with train/test split
+metadata <- data.frame(Set = c(rep("train", 7), rep("test", 3)),
+                       row.names = samples)
+
+#bundle the tables
+tables <- list(table1, table2, table3)
+
+#run joint_rpca
+result <- joint_rpca(
+  tables = tables,
+  n_components = 2,
+  n_test_samples = 3,
+  sample_metadata = metadata,
+  train_test_column = "Set",
+  rclr_transform_tables = TRUE,
+  min_sample_count = 0,
+  min_feature_count = 0,
+  min_feature_frequency = 0,
+  max_iterations = 5
+)
+
+#view results
+print(result$ord_res$samples)
+print(result$ord_res$features)
+print(result$cv_stats)
```

Comment:
Once the work is otherwise ready we need to consider how to treat outputs but that can be omitted for the time being.